modelzoo/DIN/train.py

```diff
@@ -1,387 +1,739 @@
+            partitioner=self.dense_layer_partitioner,
+            reuse=tf.AUTO_REUSE,
+        )
+        with top_mlp_scope.keep_weights(dtype=tf.float32) if self.bf16 \
+            else top_mlp_scope:
+            self.logits = self.top_fc_layer(top_input)
+        if self.bf16:
+            self.logits = tf.cast(self.logits, dtype=tf.float32)
+
+        predict = tf.math.sigmoid(self.logits)
+        return predict
+
+    def optimizer(self):
+        self.logits = tf.squeeze(self.logits)
+        loss = tf.losses.sigmoid_cross_entropy(
+            self.label,
+            self.logits,
+            scope='loss',
+            reduction=tf.losses.Reduction.SUM_OVER_BATCH_SIZE)
+        tf.summary.scalar('loss', loss)
+
+        self.global_step = tf.train.get_or_create_global_step()
+        optimizer = tf.train.AdamOptimizer(learning_rate=self.learning_rate)
+        gradients = optimizer.compute_gradients(loss)
+        clipped_gradients = [(tf.clip_by_norm(grad, 5), var)
+                             for grad, var in gradients if grad is not None]
+
+        train_op = optimizer.apply_gradients(clipped_gradients,
+                                             global_step=self.global_step)
+
+        return train_op, loss
 
 
 def get_arg_parser():
     parser = argparse.ArgumentParser()
-    parser.add_argument('--job',
-                        help='train or test model',
-                        type=str,
-                        choices=["train", "test"],
-                        default='train')
-    parser.add_argument('--seed', help='set random seed', type=int, default=3)
     parser.add_argument('--data_location',
                         help='Full path of train data',
                         required=False,
                         default='./data')
+    parser.add_argument('--steps',
+                        help='set the number of steps on train dataset',
+                        type=int,
+                        default=0)
+    parser.add_argument('--batch_size',
+                        help='Batch size to train. Default is 512',
+                        type=int,
+                        default=128)
     parser.add_argument('--output_dir',
                         help='Full path to logs & model output directory',
                         required=False,
                         default='./result')
-    parser.add_argument('--model_dir',
-                        help='Full path to test model directory',
+    parser.add_argument('--checkpoint',
+                        help='Full path to checkpoints input/output directory',
                         required=False)
+    parser.add_argument('--deep_dropout',
+                        help='Dropout regularization for deep model',
+                        type=float,
+                        default=0.0)
+    parser.add_argument('--learning_rate',
+                        help='Learning rate for model',
+                        type=float,
+                        default=0.001)
```

Comment:
Why is learning rate so low for DIN model? How does the model behave for higher values of learning rate?

Reply:
I got better acc/auc with this learning rate compared to a higher value.